dt-backend/src/model/user/user.controller.ts

```diff
@@ -0,0 +1,34 @@
+import {Controller, Get, Post, Delete, Patch, Body, Param} from "@nestjs/common";
+import { User } from '../types/user.type'
+import { UserService} from "./user.service";
+
+@Controller('user')
+export class UserController {
+
+    constructor(private readonly userService: UserService) {}
+
+    @Post('createuser')
+    async createUser (@Body() userData: {username: string; mail: string; password: string }): Promise<User> {
+        return this.userService.createUser(userData);
+    }
+
+    @Get(":id")
+    async getUserById(@Param('id') id: string): Promise<User> {
+        return this.userService.User({id: id});
+    }
+
+    @Patch('update/:id')
+    async updateUser(@Param('id') id: string): Promise<User> {
+        return this.userService.updateUser({
+            where: { id: id },
+            data: { },
```

Comment:
The `where`, `data` are Prisma attributes, - pretty sure that shouldn't be here!  
Within the controller you should just pass through the input as you get it or with minor changes to the service, which then parses and handles the data! But this looks unfinished either way as `data` is empty, but just so you know how stuff should be ^^! The whole request misses the `@Body()`. I'll comment on this later again if/when you update it ^^!